src/core/vectortile/qgsvectortilelayer.cpp

```diff
@@ -694,3 +704,41 @@ QgsVectorTileLabeling *QgsVectorTileLayer::labeling() const
 {
   return mLabeling.get();
 }
+
+
+
+//
+// QgsAnnotationLayerDataProvider
+//
+///@cond PRIVATE
+QgsVectorTileDataProvider::QgsVectorTileDataProvider(
+  const ProviderOptions &options,
+  QgsDataProvider::ReadFlags flags )
+  : QgsDataProvider( QString(), options, flags )
+{}
+
+QgsCoordinateReferenceSystem QgsVectorTileDataProvider::crs() const
+{
+  return QgsCoordinateReferenceSystem();
```

Comment:
BTW, you might as well return QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:3857" ) ), that's a limitation of the vector tile reader which allows us to return a constant CRS: )